internal/install/version_file.go

```diff
@@ -9,22 +9,27 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/pkg/errors"
 
 	"github.com/elastic/elastic-package/internal/configuration/locations"
 	"github.com/elastic/elastic-package/internal/version"
 )
 
-func checkIfLatestVersionInstalled(elasticPackagePath string) (bool, error) {
-	versionFile, err := ioutil.ReadFile(filepath.Join(elasticPackagePath, versionFilename))
+func checkIfLatestVersionInstalled(elasticPackagePath *locations.LocationManager) (bool, error) {
+	versionPath := filepath.Join(elasticPackagePath.RootDir(), versionFilename)
+	versionFile, err := ioutil.ReadFile(versionPath)
 	if os.IsExist(err) {
 		return false, nil // old version, no version file
 	}
 	if err != nil {
 		return false, errors.Wrap(err, "reading version file failed")
 	}
 	v := string(versionFile)
+	if version.CommitHash == "undefined" && strings.Contains(v, "undefined") {
+		fmt.Printf("WARNING: CommitHash is undefined, in both %s and the compiled binary, config may be out of date, and elastic-package was not properly built.\n", versionPath)
```

Comment:
Let's use `logger.Warnf` here, instead of `fmt.Printf`.

Also, I wouldn't say the last bit about `elastic-package` not being built properly. In our README we allow users to install `elastic-package` using `go get`, and with this method the commit hash is undefined but the package is still built properly otherwise.

Reply:
Yah, agreed. Gonna remove that part.